docs/proposals/promote-dependencies-automatically/README.md

```diff
@@ -0,0 +1,196 @@
+Typically, users don't create Pods directly. Instead, users create them using workload resources such as Deployment, StatefulSet, or Job. Take Deployment as an example. As shown in the following figure, the dependent resources of Deployment include ConfigMaps, Secrets, PVC, and Services.
+
+And, in addition to the native resources in kubernetes, the resources on which the user's CRD/CR resources depend need to support automatic promotion.
+
+![dependencies](statics/dependenciesOfDeployment.drawio.svg)
+
+## Motivation
+
+Currently, `karmadactl promote` only supports promotion of resources one by one. When promoting workload resources through Karmada, users need to manually promote their dependent resources. Maintaining the dependent resources of the target cluster becomes a heavy burden.
+
+### Goals
+
+Provide a strategy to automatically promote the dependent resources of a target workload when promoting the target workload resources to be managed by Karmada.
+
+### Non-Goals
+
+Support enabling and disabling this feature, with the default setting as enabled.
+
+## Proposal
+
+This proposal introduces a strategy to `karmadactl`, whereby when users promote resources, all the resources that the promoted resource depends on will also be automatically promoted to be managed by Karmada.
+
+### Function overview
+
+1. Support automatic promotion of dependent resources for native Kubernetes resources.
+2. Support automatic promotion of dependent resources for CR(Custom Resource).
+
+### User Stories (Optional)
+
+#### Story 1
+
+Suppose there is a member cluster where a workload (such as Deployment or custom resource) is deployed, but it is not managed by Karmada. This resource references ConfigMaps and Secrets.
+
+- Action：
+  - The user use the `karmadactl promote` command to promote this resource and let Karmada take over the management of this workload.
+- Result：
+  - The ConfigMaps and Secrets that the resource depends on will also be promoted to be managed by Karmada.
+
+## Design Details
+
+### Architecture
+
+![architecture](statics/architecture.drawio.svg)
+From the above architecture diagram, the workflow for promoting the resource and its dependent resources is as follows:
+1. Retrieve the ObjInfo(Object Information) of the workload to be promoted from the member cluster.
+2. Determine whether to enable the functionality of automatically promoting all dependent resources based on the value of `Deps`. 
+   1. If `Deps` is set to `false`, only create the corresponding Resource Template, ClusterPropagationPolicy/PropagationPolicy, and generate the respective ResourceBinding in the Karmada control plane for the workload.
+   2. If `Deps` is set to `true`, use the interpret framework to analyze the dependencies of the corresponding workload. Create the Resource Templates, ClusterPropagationPolicy/PropagationPolicy, and ResourceBindings for the dependencies in the Karmada control plane. Then, create the ResourceBinding for the workload itself. 
```

Comment:
I think the `karmadactl promote` doesn't take responsibility to create the `ResourceBinding`, right? It just focuses on promoting the dependents of `resource-templates`.

Reply:
Yes.